src/Features/Core/Portable/Diagnostics/EngineV2/ICompilerDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Workspaces.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
+{
+    internal interface ICompilerDiagnosticAnalyzer : IWorkspaceService
```

Comment:
Name is confusing with [CompilerDiagnosticAnalyzer](http://source.roslyn.io/Microsoft.CodeAnalysis/R/39619341c8ee7e95.html) that lies in the compiler layer for reporting pure compiler/build diagnostics. Can we instead rename this to something like IDiagnosticAnalyzerDriver and add a comment that is the driver for executing pure diagnostic analyzers (i.e. need no workspace context) using CompilationWithAnalyzers?